primitives/domain/src/lib.rs

```diff
@@ -46,17 +46,47 @@ impl Display for McEpochNumber {
 		u32::fmt(&self.0, f)
 	}
 }
-/// Amount of Lovelace (which is a fraction of 1 ADA) staked/locked on Cardano
+
 #[derive(Default, Clone, Copy, Debug, Encode, Decode, TypeInfo, ToDatum, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(Serialize))]
-pub struct StakeDelegation(pub u64);
+pub struct StakeDelegation {
+	/// Amount of Lovelace (which is a fraction of 1 ADA) staked/locked on Cardano
+	pub ada: StakeAmount,
+
+	// TODO ETH: u64 if not enough to count stake on Ethereum in Wei
+	/// Amount of Wei (which is a fraction of 1 ETH) staked/locked on Ethereum
+	pub eth: StakeAmount,
```

Comment:
for later: better to keep these as separate types

Reply:
Introduced EthStakeAmount type